include/dlaf/matrix/internal/tile_future_manager.h

```diff
@@ -0,0 +1,220 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+#include <hpx/hpx.hpp>
+#include "dlaf/matrix/tile.h"
+
+namespace dlaf {
+namespace matrix {
+namespace internal {
+
+template <class ReturnTileType, class TileType>
+hpx::future<ReturnTileType> setPromiseTileFuture(hpx::future<TileType>& old_future,
+                                                 hpx::promise<TileType>& p) noexcept {
+  return old_future.then(hpx::launch::sync, [p = std::move(p)](hpx::future<TileType>&& fut) mutable {
+    try {
+      return ReturnTileType(std::move(fut.get().setPromise(std::move(p))));
```

Comment:
`fut.get().setPromise(std::move(p))` this is a rvalue, no need for the outer std::move

Reply:
setPromise() returns a reference to a Tile, so it is a lvalue.

Reply:
I meant `... std::move(fut.get() ... )`